vault_core/src/vault.rs

```diff
@@ -554,20 +554,21 @@ impl Vault {
             return Ok(0);
         }
 
-        let fee = vrt_rewards
-            .checked_mul(self.reward_fee_bps() as u64)
+        let fee = (vrt_rewards as u128)
```

Comment:
you can also do try into here

let fee = (vrt_rewards as u128)
    .checked_mul(self.reward_fee_bps() as u128)
    .ok_or(VaultError::VaultOverflow)?
    .div_ceil(10_000)
    .try_into()
    .map_err(|_| VaultError::VaultOverflow)?;